### PR TITLE
Update omnibus to function with Solaris 11 natively

### PR DESCRIPTION
**Pull request by scotthain**
This puts some Solaris 11 specific functionality in the omnibus cookbook.

### CODE REVIEW

File: libraries/helper.rb [hunk 1]
```diff
@@ -35,6 +35,8 @@ def build_user_home
         File.join('/Users', node['omnibus']['build_user'])
       elsif windows?
         windows_safe_path_join(ENV['SYSTEMDRIVE'], 'Users', node['omnibus']['build_user'])
+      elsif node['platform_family'] == 'solaris2'
```
Comment (schisamo) [inline libraries/helper.rb]: Can we use the `solaris2?` Sugar helper here?

Comment (scotthain) [inline libraries/helper.rb]: I had that originally, specs failed all over the place... Might need to update Sugar a bit? I've never delved into its code.

File: recipes/_common.rb [hunk 2]
```diff
@@ -28,6 +28,14 @@
   recursive true
 end
 
+# If we are on Solaris 11, we need to update some paths to favor the gnu utils
+if node['platform_family'] == 'solaris2' && node['platform_version'].to_f == 5.11
+  unless ENV['PATH'].include? '/usr/gnu/bin'
```
Comment (schisamo) [inline recipes/_common.rb]: try:

```ruby
if solaris2? && version(node['platform_version']).satisfies?('>= 5.11')
```
or
```ruby
# Add this require to `omnibus::_common`
require 'chef/sugar/core_extensions'
if solaris2? && node['platform_version'].satisfies?('>=  5.11')
```

**Comment by schisamo** [comment]
@scotthain Added a few comments. What is the status of related PR https://github.com/opscode-cookbooks/omnibus/pull/75?

File: recipes/_common.rb (hunk 2 above)
Comment (scotthain) [inline recipes/_common.rb]: Check what I had before: https://github.com/opscode-cookbooks/omnibus/commit/33a5cdc6f60d7046a3595757ccb7d6a32d2b065c

**Comment by scotthain** [comment]
#75 is actually unrelated to this, and will be picked up/refactored/removed (whatever's appropriate) once the omnibus-build-essential project goes live.

**Pull request merged**